graphs/adapter-jgrapht/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/jgrapht/check/GetSimpleCycleCountCheck.kt

```diff
@@ -0,0 +1,23 @@
+package com.anaplan.engineering.azuki.graphs.adapter.jgrapht.check
+
+import com.anaplan.engineering.azuki.core.system.LateDetectUnsupportedCheckException
+import com.anaplan.engineering.azuki.graphs.adapter.api.GetCycleCountBehavior
+import com.anaplan.engineering.azuki.graphs.adapter.jgrapht.execution.ExecutionEnvironment
+import org.jgrapht.alg.cycle.HawickJamesSimpleCycles
+
+class GetSimpleCycleCountCheck(
+    private val graphName: String,
+    private val result: Long
+) : JGraphTCheck, GetCycleCountBehavior() {
+
+    override fun check(env: ExecutionEnvironment) =
+        checkEqual(result, env.get<String, Long>(graphName) {
+            if (type.isDirected) {
+                HawickJamesSimpleCycles(this).findSimpleCycles().size.toLong()
```

Comment:
⛏️ Calling `countSimpleCycles` was perfectly fine here. All I wanted to say with https://github.com/anaplan-engineering/azuki/pull/40#discussion_r1576064082 was that just because the other cycle finder implementations don't have an explicit _count_ method, it doesn't mean that you can't easily use them to count cycles.